### PR TITLE
feat(highlight): Allow highlighting multiple terms in one go

### PR DESCRIPTION
**Pull request by bartoszbobin**
#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/barista-components/highlight/src/highlight.ts [hunk 1]
```diff
@@ -64,6 +64,13 @@ function escapeRegExp(text: string): string {
   return text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 }
 
+function createTermsRegExp(terms: string[], caseSensitive: boolean): RegExp {
```
Comment (Sherif-Elhefnawy) [inline libs/barista-components/highlight/src/highlight.ts:67]: Please add jsDoc comment for this new function

File: libs/examples/src/highlight/highlight-term-array-example/highlight-term-array-example.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<input
+  type="text"
+  dtInput
+  value="sit,est,lacus massa"
+  #search
+  aria-label="Split terms by ','"
+/>
+<dt-highlight [term]="search.value.split(',')" caseSensitive>
+  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi id ultrices
+  eros. Integer viverra tempor lacinia. Aliquam ac nibh consectetur, auctor
```
Comment (Sherif-Elhefnawy) [inline libs/examples/src/highlight/highlight-term-array-example/highlight-term-array-example.html:10]: You need to add this file declaration in `libs/examples/src/index.ts` as well as add a routing link in `apps/demos/src/app-routing.module.ts`

**Review by Sherif-Elhefnawy — CHANGES_REQUESTED** [review]

**Comment by Sherif-Elhefnawy** [comment]
Running `npm run format` should help you with the formatting check error, there seems to be an issue with `highlight.spec.ts`

File: libs/barista-components/highlight/README.md [hunk 3]
```diff
@@ -55,3 +55,15 @@ be escaped! You have to avoid any HTML Tags inside this component!
 ### A case sensitive highlight example
 
 <ba-live-example name="DtExampleHighlightCaseSensitive"></ba-live-example>
+
+### Multiple terms
+
+```html
```
[the remark this answers is not in the document]
Comment (bartoszbobin) [reply libs/barista-components/highlight/README.md]: Done

**Comment by bartoszbobin** [comment]
> Running `npm run format` should help you with the formatting check error, there seems to be an issue with `highlight.spec.ts`

Unfortunately 46 files have been changed (mostly specs) but not `highlight.spec.ts`

**Comment by Sherif-Elhefnawy** [comment]
Hi, I pushed changes for formatting on this commit/branch that should help with the issue with the highlight.spec.ts https://github.com/dynatrace-oss/barista/commit/895656dd1bc6e54c8a09082471a5a72ccc7e7554#diff-d14ea366c314794bd31fe9e4b765acbb7efb5f815ec9136bbfb552cd0350ae59

File: libs/examples/src/highlight/highlight-term-array-example/highlight-term-array-example.html (hunk 2 above)
Comment (Sherif-Elhefnawy) [reply libs/examples/src/highlight/highlight-term-array-example/highlight-term-array-example.html]: Please don't forget to add the nav link to` app-routing.module.ts` and `nav-items.ts` as was done here: https://github.com/dynatrace-oss/barista/pull/2596/files#diff-ffeda48d19c6203493253e37af0053e5732bc3952ede6423d1f5544919939abc

**Comment by Sherif-Elhefnawy** [comment]
@bartoszbobin, thank you for fixing the formatting issue you had, please don't forget to add the nav link to app-routing.module.ts and nav-items.ts as was done here for your example: https://github.com/dynatrace-oss/barista/pull/2596/files#diff-ffeda48d19c6203493253e37af0053e5732bc3952ede6423d1f5544919939abc

**Comment by Sherif-Elhefnawy** [comment]
Thank you for adding the navigation, now please squash all 3 commits into one commit with the appropriate message for the feature :)

**Review by Sherif-Elhefnawy — APPROVED** [review]

**Review by Sherif-Elhefnawy — APPROVED** [review]

**Comment by Sherif-Elhefnawy** [comment]
Cherry-picked into: 
10.8.x: 3d346c1435cbc2b291506c6ccd37d3f806270ac8
master: 60dba64a2d45fb6e93aa5bf2907bd3746524f133

**Pull request merged**